scripts/word_embeddings/extract_vocab.py

```diff
@@ -0,0 +1,94 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=
+"""Extract the vocabulary from a file and write it to disk."""
+
+import argparse
+import itertools
+import json
+import logging
+import time
+
+import gluonnlp as nlp
+
+
+def parse_args():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(
+        description='Vocabulary extractor.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--max-size', type=int, default=None)
+    parser.add_argument('--min-freq', type=int, default=5)
+    parser.add_argument('--max-word-length', type=int, default=50)
+    parser.add_argument('files', type=str, nargs='+')
+    parser.add_argument('--vocab-output', type=str, default='vocab.json')
+    parser.add_argument('--counts-output', type=str, default='counts.json')
+    args = parser.parse_args()
+    return args
+
+
+def get_vocab(args):
```

Comment:
Is this script supposed to be used once for preprocessing of vocab? Do we want to put it under /scripts?

Reply:
Yes. But the dataset changes weekly so we currently don't distribute a preprocessed version of the dataset.

Reply:
The dataset changes weekly?

Reply:
Yes, its just a dump of wikipedia: https://dumps.wikimedia.org/enwiki/